CRM/Contact/Import/Parser/Contact.php

```diff
@@ -2162,4 +2139,51 @@ public function checkRelatedContactFields($relKey, $params) {
     return $allowToCreate;
   }
 
+  /**
+   * Get the possible contact matches.
+   *
+   * 1) the chosen dedupe rule falling back to
```

Comment:
This check is falling back to the default "Unsupervised" rule not the Rule chosen in the Import UI.  I added:  $checkParams['dedupe_rule_id'] = $this->_dedupeRuleGroupID ?? NULL; to get things to work on my end.  

Is that the way this is supposed to work?  The import code checks for dupes a number of times.  But I've been struggling with inconsistent results.   As my importing is based off of my own custom rules.  

I can do a PR if indeed the selected Rule should run at this point.

Reply:
@darrick I've been doing some clean up on this code over this release & last - I'm happy to look into this & put up a PR if you are OK to test it - it's a bit of a moving target - you can see that this was just merged https://github.com/civicrm/civicrm-core/pull/23455 allowing code removal in  https://github.com/civicrm/civicrm-core/pull/23461

Reply:
Okay.  I'll check that out.  It's the weekend and I should stop working.  I made two tests here: https://github.com/darrick/civicrm-core/commit/9b1470a89baee225e6f03206d02170afa8c27b23
 the second test fails when external_identifier doesn't match but Dedupe does match.  The first test passes when not external_identifier is passed in.  My current use case is trying to maintain sync with external data even after contacts are merged on the Civi side.

Reply:
oh nice - I guess the thing that will change with the way things are going is that we would use `getSubmittedValue('dedupe_rule_id')` to do the rule ID - but I'd also quite like to combine the 2 separate places where we do do dedupe look ups -

Yes - get some weekend in!

Reply:
@darrick your test + some additional clean up is in this PR https://github.com/civicrm/civicrm-core/pull/23473 - I also did a version with even more cleanup - https://github.com/civicrm/civicrm-core/pull/23476 - if you have the capacity to take a look & confirm we can get them merged